src/view/components/microbit/MicrobitSimulator.tsx

```diff
@@ -170,9 +175,56 @@ export class MicrobitSimulator extends React.Component<any, IState> {
         event.preventDefault();
         this.handleButtonClick(key, true);
     };
-
     protected onMouseLeave = (event: Event, key: string) => {
         event.preventDefault();
         console.log(`To implement onMouseLeave ${key}`);
     };
+    protected onKeyEvent(event: KeyboardEvent, active: boolean, key: string) {
+        event.stopPropagation();
+        if ([event.code, event.key].includes(CONSTANTS.KEYBOARD_KEYS.ENTER)) {
+            this.handleButtonClick(key, active);
+            if (this.imageRef.current) {
+                if (key === BUTTONS_KEYS.BTN_A) {
+                    this.imageRef.current.updateButtonAttributes(
+                        BUTTONS_KEYS.BTN_A,
+                        active
+                    );
+                } else if (key === BUTTONS_KEYS.BTN_B) {
+                    this.imageRef.current.updateButtonAttributes(
+                        BUTTONS_KEYS.BTN_B,
+                        active
+                    );
+                } else if (key === BUTTONS_KEYS.BTN_AB) {
+                    this.imageRef.current.updateButtonAttributes(
+                        BUTTONS_KEYS.BTN_AB,
+                        active
+                    );
+                }
+            }
+        } else if (
+            [event.code, event.key].includes(CONSTANTS.KEYBOARD_KEYS.A)
+        ) {
+            this.handleButtonClick(BUTTONS_KEYS.BTN_A, active);
+            if (this.imageRef.current) {
+                this.imageRef.current.updateButtonAttributes(
+                    BUTTONS_KEYS.BTN_A,
+                    active
+                );
+            }
+        } else if (
+            [event.code, event.key].includes(CONSTANTS.KEYBOARD_KEYS.B)
+        ) {
+            this.handleButtonClick(BUTTONS_KEYS.BTN_B, active);
+            if (this.imageRef.current) {
+                this.imageRef.current.updateButtonAttributes(
+                    BUTTONS_KEYS.BTN_B,
+                    active
+                );
+            }
+        } else if (event.key === CONSTANTS.KEYBOARD_KEYS.CAPITAL_F) {
+            this.togglePlayClick();
+        } else if (event.key === CONSTANTS.KEYBOARD_KEYS.CAPITAL_R) {
+            this.refreshSimulatorClick();
+        }

```

Comment:
Refreshing also seems like it's doing things more than once. Here are my logs after pressing `R` once:
```
[INFO] Deploying code to the simulator...

[INFO] Deploying code to the simulator...
[INFO] File selected : d:\microbit\python_test\code.py 
[INFO] File selected : d:\microbit\python_test\code.py 

[INFO] Deploying code to the simulator...

[INFO] Deploying code to the simulator...
[INFO] File selected : d:\microbit\python_test\code.py 
[INFO] File selected : d:\microbit\python_test\code.py 
```

Also if I press `R` multiple times the simulator goes funky:
![funky_example](https://user-images.githubusercontent.com/12337821/75470216-3250e800-5945-11ea-9b21-d1dde897fe31.gif)

Reply:
Thanks to the lovely Kevin for fixing this! It is fixed :D